> > /media/disk1/braddr/sandbox/at-client/pull-4072458-Linux_64_64/src/dmd/src/src/dmd/mtype.h:341:5: error: ‘TypeMixin’ does not name a type

That means no one has added it to the Visitor class either then...

--- a/src/dmd/module.h
+++ b/src/dmd/module.h
@@ -101,12 +101,12 @@ public:
     Modules aimports;             // all imported modules
 
     unsigned debuglevel;        // debug level
-    Strings *debugids;      // debug identifiers
-    Strings *debugidsNot;       // forward referenced debug identifiers
+    Identifiers *debugids;      // debug identifiers
+    Identifiers *debugidsNot;   // forward referenced debug identifiers
 
     unsigned versionlevel;      // version level
-    Strings *versionids;    // version identifiers
-    Strings *versionidsNot;     // forward referenced version identifiers
+    Identifiers *versionids;    // version identifiers
+    Identifiers *versionidsNot; // forward referenced version identifiers
 
     MacroTable macrotable;      // document comment macros
     Escape *escapetable;        // document comment escapes

--- a/src/dmd/mtype.h
+++ b/src/dmd/mtype.h
@@ -338,6 +338,8 @@ public:
     TypeTuple *isTypeTuple();
     TypeSlice *isTypeSlice();
     TypeNull *isTypeNull();
+    TypeMixin *isTypeMixin();
+    TypeTraits *isTypeTraits();
 
     void accept(Visitor *v) { v->visit(this); }
 };

--- a/src/dmd/mtype.h
+++ b/src/dmd/mtype.h
@@ -651,6 +651,17 @@ class TypeTraits : public Type
     void accept(Visitor *v) { v->visit(this); }
 };
 
+class TypeMixin : public Type
+{
+    Loc loc;
+    Expressions *exps;
+
+    const char *kind();
+    Type *syntaxCopy();
+    Dsymbol *toDsymbol(Scope *sc);
+    void accept(Visitor *v) { v->visit(this); }
+};
+
 class TypeQualified : public Type
 {
 public:

--- a/src/dmd/parsetimevisitor.d
+++ b/src/dmd/parsetimevisitor.d
@@ -147,6 +147,7 @@ public:
     void visit(AST.TypeNext t) { visit(cast(AST.Type)t); }
     void visit(AST.TypeQualified t) { visit(cast(AST.Type)t); }
     void visit(AST.TypeTraits t) { visit(cast(AST.Type)t); }
+    void visit(AST.TypeMixin t) { visit(cast(AST.Type)t); }
 
     // TypeNext
     void visit(AST.TypeReference t) { visit(cast(AST.TypeNext)t); }

--- a/src/dmd/strictvisitor.d
+++ b/src/dmd/strictvisitor.d
@@ -121,6 +121,7 @@ extern(C++) class StrictVisitor(AST) : ParseTimeVisitor!AST
     override void visit(AST.TypeSArray) { assert(0); }
     override void visit(AST.TypeQualified) { assert(0); }
     override void visit(AST.TypeTraits) { assert(0); }
+    override void visit(AST.TypeMixin) { assert(0); }
     override void visit(AST.TypeIdentifier) { assert(0); }
     override void visit(AST.TypeReturn) { assert(0); }
     override void visit(AST.TypeTypeof) { assert(0); }

--- a/src/dmd/transitivevisitor.d
+++ b/src/dmd/transitivevisitor.d
@@ -464,6 +464,11 @@ package mixin template ParseVisitMethods(AST)
         t.exp.accept(this);
     }
 
+    override void visit(AST.TypeMixin t)
+    {
+        visitArgs(t.exps);
+    }
+
 //      Miscellaneous
 //========================================================
 

--- a/src/dmd/visitor.h
+++ b/src/dmd/visitor.h
@@ -82,6 +82,7 @@ class TypeTuple;
 class TypeSlice;
 class TypeNull;
 class TypeTraits;
+class TypeMixin;
 
 class Dsymbol;
 
@@ -431,6 +432,7 @@ public:
     virtual void visit(TypeNext *t) { visit((Type *)t); }
     virtual void visit(TypeQualified *t) { visit((Type *)t); }
     virtual void visit(TypeTraits *t) { visit((Type *)t); }
+    virtual void visit(TypeMixin *t) { visit((Type *)t); }
 
     // TypeNext
     virtual void visit(TypeReference *t) { visit((TypeNext *)t); }

